packages/js/src/ui/components/primitives/Popover/PopoverContent.tsx

```diff
@@ -41,37 +41,46 @@ type PopoverContentProps = JSX.IntrinsicElements['div'] & { appearanceKey?: Appe
 export const PopoverContent = (props: PopoverContentProps) => {
   const { open, onClose, reference, floating } = usePopover();
   const { active } = useFocusManager();
+  const { containerElement } = useAppearance();
 
-  const handleClickOutside = (e: MouseEvent) => {
+  const handleClickOutside: EventListener = (e) => {
     // Don't count the trigger as outside click
     if (reference()?.contains(e.target as Node)) {
       return;
     }
 
-    if (active() !== floating() || floating()?.contains(e.target as Node)) {
+    const container = containerElement();
+
+    if (
+      active() !== floating() ||
+      floating()?.contains(e.target as Node) ||
+      (container && (e.target as Element).shadowRoot === container)
```

Comment:
if the target shadow root is the container where the inbox is embedded, then do not close it;